lib/buttons-and-links/KButton.vue

```diff
@@ -22,11 +22,11 @@
       data-test="iconBefore"
       class="prop-icon"
     />
-
+    <slot name="menu"></slot>
     <!-- @slot Pass sub-components into the button, which provides more flexibility than and takes precedence over the `text` prop -->
     <slot v-if="$slots.default"></slot>
 
-    <template v-else>
```

Comment:
Previously, the component was structured to either have slots or text, but not both. My thinking was that there could be scenarios where some combination of text and slots would be necessary. For example, in Kolibri, we have one place where there is a secondary button "Options" with a dropdown menu.
<img width="821" alt="Screen Shot 2022-09-01 at 3 23 25 PM" src="https://user-images.githubusercontent.com/17235236/187995796-7989db1c-4cfe-4b43-93e4-affc68872cfd.png">
I made this change thinking about this use case, which would allow both the `text` on line 30 to remain as well as the `<slots>`. There might be a better way to do this, though!

Reply:
Thank you for the explanation, I think that's fine as long as it's communicated in the documentation, which means that this comment needs to be updated as the default slot doesn't take precedence over the `text` prop anymore
```
<!-- @slot Pass sub-components into the button, which provides more flexibility than and takes precedence over the `text` prop -->
```

and it would also ideally be mentioned in the changelog.